src/db.c

```diff
@@ -85,7 +129,7 @@ void updateLFU(robj *val) {
  * expired on replicas even if the master is lagging expiring our key via DELs
  * in the replication link. */
 robj *lookupKey(redisDb *db, robj *key, int flags) {
-    dictEntry *de = dictFind(db->dict,key->ptr);
+    dictEntry *de = dictFind(getDict(db, key->ptr),key->ptr);
```

Comment:
Apologize that I didn't read the PR description. Great to hear there's no perf downgrade.

Reply:
I'm working on finding a way to get rid of extra hash calculations. There is no clear solution just yet, but I plan on addressing it either as part of this PR or short after, depending on the mechanism that we choose and whether or not it requires larger refactoring.

Reply:
@zuiderkwast @madolson @molian6 
I think I might have found a way to avoid redundant CRC hash calculations, for that we could add a separate field into redisDb that would hold a reference to the current slot specific dictionary. We could populate it in the same place where we determine current command's slot. https://github.com/redis/redis/blob/unstable/src/server.c#L3876
It should work as long as entire command operates only on one slot, which is a case currently.
This would save us from doing a lot of plumbing, and pushing slot ID around through db.c APIs. WDYT?